src/tests/SearchView.test.jsx

```diff
@@ -345,23 +375,37 @@ describe('Search View', () => {
     const searchInput = getByRole('textbox');
 
     await user.type(searchInput, 'grahamchapman@python.com');
+    jest.runOnlyPendingTimers();
 
     expect(spyOnFetch).toHaveBeenCalledWith(
       'https://treeherder.mozilla.org/api/project/autoland/push/?author=grahamchapman@python.com',
     );
 
-    expect(store.getState().search.errorMessage).toBe(
+    await waitFor(() =>
+      expect(queryByText('She turned me into a newt')).toBeInTheDocument(),
+    );
+
+    expect(store.getState().search.inputError).toBe(true);
+    expect(store.getState().search.inputHelperText).toBe(
       'She turned me into a newt',
     );
   });
 
   it('should not call fetch if searchValue is not a hash or email', async () => {
-    const spyOnFetch = jest.spyOn(global, 'fetch');
-    const user = userEvent.setup();
+    const spyOnFetch = jest.spyOn(global, 'fetch').mockImplementation(() =>
+      Promise.resolve(
+        Promise.resolve({
```

Comment:
I believe you want something such as:
```js
Promise.resolve({
  json: () => Promise.resolve({ results: [] }),
});
```

I see you used the same structure elsewhere, so feel free to fix this in the other locations too.

Let me also explain what `Promise.resolve` does in a few lines. `Promise.resolve(value)` will:
* if `value` isn't a promise (or rather a "thenable", that is an object with a "then" method and some properties, but let's keep it simple) => this returns a resolved promise with the value as the resolved value
* if `value` is a promise => this returns a promise with the same status as the value. This means that:
  * If the input promise is resolved, then the returned promise is resolved with the same resolved value
  * If the input promise is rejected, then the returned promise is rejected with the same rejected value
  * If the input promise is still "pending", then the returned promise is also pending, and will be resolved/rejected whenever the input promise is also resolved/rejected.

This means that constructs like `Promise.resolve(Promise.resolve(something))` or `Promise.resolve(Promise.reject(something))` are redundant.

Therefore `Promise.resolve` is useful when you need to make a promise value from an input value that's not a promise, or that could either be a promise or a non-promise. Also note that `await` does the exact same thing with the value at the right of `await`.

Hope this clears things up :-)

Reply:
I think I just had a headache trying to get the tests to work :sweat_smile: removing the extra resolve did work fine